File: 136.只出现一次的数字.py
#
# @lc app=leetcode.cn id=136 lang=python3
#
# [136] 只出现一次的数字
#

# @lc code=start
from typing import List


class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        # return 2*sum(set(nums)) - sum(nums)
        res = 0
        for i in nums:
            res ^= i
        return res
# @lc code=end

File: 136.只出现一次的数字.java
/*
 * @lc app=leetcode.cn id=136 lang=java
 *
 * [136] 只出现一次的数字
 */

// @lc code=start
class Solution {
    public int singleNumber(int[] nums) {
        int res = 0;
        for (int n : nums)
            res = res ^ n;
        return res;
    }
}
// @lc code=end

